packages/moonstone/internal/DateTimeDecorator/DateTimeDecorator.js

```diff
@@ -169,17 +149,39 @@ const DateTimeDecorator = hoc((config, Wrapped) => {
 		}
 
 		emitChange = (date) => {
-			const {onChange} = this.props;
-			if (onChange) {
-				onChange({
-					value: date ? date.getJSDate() : null
-				});
+			forward('onChange', {value: date ? date.getJSDate() : null}, this.props);
+		}
+
+		handleOpen = (ev) => {
+			forward('onOpen', ev, this.props);
+
+			const newValue = toTime(this.props.value);
+			const value = newValue || Date.now();
+
+			// if we're opening, store the current value as the initial value for cancellation
+			this.setState({
```

Comment:
Unrelated to this, I noticed that cancellation doesn't work any longer for DatePicker and TimePicker. I haven't bisected to find the root cause yet but we should track it down since without it storing the initial value is not productive.

Reply:
Let's create a ticket for the bug. It seems like it's been broken for a very long time.